### ISSUE TITLE
No warnings for a loop with byte indexer and Int32 condition.

### ISSUE THREAD

**Issue by EgorBo** [dotnet/roslyn#36270]
**Version Used**: 
3.100.19.26603 (9d80dea7).
Also tried on "master" at [sharplab.io](https://sharplab.io/#v2:EYLgtghgzgLgpgJwD4AEBMBGAsAKHQAgAV9cBvXfS/FDANmoBZ8BZCASwDsAKASgqvI4qw/JxgBtALr54sfAF58HOAHdRHCQGYADNskBufiPwAzAPYJ8XYAE94ohfm36HAHhlxYAOgAycDgDmMAAWLmwA1OF8QsYiNACcXGw8hjGUAL646UA)

**Steps to Reproduce**:
1. Compile the following code:
```csharp
using System;
class P 
{
    static void Main()
    {
        int[] test = new int[300];
        for (byte i = 0; i < test.Length; i++)
            Console.WriteLine(i);
    }
}
```
**Expected Behavior**:
a warning e.g.: 
```
test.Length is int and can be outside the range of type 'byte'
```

**Actual Behavior**:
No warnings generated. The loop is endless.

However, the following code does generate a warrning:
```csharp
for (byte i = 0; i < 300; i++) {}
```
```
warning CS0652: Comparison to integral constant is useless; 
the constant is outside the range of type 'byte'
```

Sorry if there is already an issue for this but I couldn't find any.

**Comment by gafter** [comment]
Are you suggesting there should be a warning whenever comparing a value of type `byte` (for example `i`) to a value of type `int` (for example `test.Length`) using `<`?  That seems like it would frequently be a false positive.  Perhaps your increment should be `checked(i++)`?

**Comment by EgorBo** [comment]
@gafter hm.. I was told that C/C++ compilers actually warn about such cases but it turns out they don't (just checked). I created this feature request due to https://github.com/mono/mono/issues/14935#issuecomment-500325931
